File: 2019/day1/day1.js
const readline = require("readline");
const fs = require("fs");

let fuelCalc = (fuel) => Math.floor((fuel / 3) - 2)
let modules = []

let rd = readline.createInterface({
    input: fs.createReadStream("input.txt"),
});

rd.on('line', (line) => {
    // Parse numbers from each line in the input file
    modules.push(parseInt(line));
}).on('close', () => {
    // Part 1
    let total = 0;
    modules.forEach((val) => {
        total += fuelCalc(val);
    });

    // Part 2
    console.log("Total Modules Fuel: ", total);
    total = 0;
    modules.forEach((val) => {
        let fuel = Math.floor((val / 3) - 2);
        while (fuel > 0) {
            total += fuel;
            fuel = Math.floor((fuel / 3) - 2);
        }
    });
    console.log("Total with Fuel's fuel:", total);
});

File: 2019/day1/day1.go
package main

import (
	"bufio"
	"log"
	"os"
	"strconv"
)

func main() {
	part1()
	part2()
}

func part1() {
	file, err := os.Open("input.txt")
	if err != nil {
		os.Exit(-1)
	}
	defer file.Close()

	total := 0
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		val, err := strconv.Atoi(scanner.Text())
		if err != nil {
			log.Fatal(err)
		}
		total += val / 3 - 2
	}
	log.Println("Part 1 - Module Fuel:", total)

}

func part2() {
	file, err := os.Open("input.txt")
	if err != nil {
		os.Exit(-1)
	}
	defer file.Close()

	total := 0
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		val, err := strconv.Atoi(scanner.Text())
		if err != nil {
			log.Fatal(err)
		}

		fuel := val / 3 - 2
		for fuel > 0 {
			total += fuel
			fuel = fuel / 3 - 2
		}
	}
	log.Println("Part 2 - Fuel's fuel:", total)
}
